### ISSUE TITLE
Using event.preventDownshiftDefault = true triggers React warning

### ISSUE THREAD

**Issue by jonavila** [paypal/downshift#404]
* `downshift` version: 1.31.2
* `node` version: 8.9.4
* `npm` (or `yarn`) version: npm 5.8.0

Relevant code or config

```javascript
{...getItemProps({
  onClick: event => {
    event.preventDownshiftDefault = true;
  },
  item,
  label: item,
})}
```

What you did:
I was preventing the default downshift handler from executing.

What happened:
React triggered a warning

index.js:2178 Warning: This synthetic event is reused for performance reasons. If you're seeing this, you're adding a new property in the synthetic event object. The property is never released. See https://fb.me/react-event-pooling for more information.
Reproduction repository:

Problem description:
Functionality works as expected, but not sure if the warning can be prevented.
Suggested solution:
If warning is normal for Dev mode, add a note to docs.

**Comment by kentcdodds** [comment]
Dang. I hadn't considered this and I'm surprised I didn't notice this. We'll have to think about another way to implement this feature without having the warning. Perhaps check `event.nativeEvent.preventDownshiftDefault` 🤔

**Comment by jonavila** [comment]
@kentcdodds  I thought to myself: "The feature has been out for almost a month. I'm must doing something wrong, hahaha". BTW, Downshift is awesome. I've just been getting familiar with it in the past day or so. I'll dig up ticket on why event.preventDefault is not a good option.

**Comment by austintackaberry** [comment]
Has this issue been resolved?

**Comment by kentcdodds** [comment]
This will be resolved in downshift@2.0.0 which has had a beta release. Official release coming soon! Please try the beta! Learn more in #461
